16/umbraco-cms/customizing/property-editors/property-value-converters.md

```diff
@@ -160,46 +180,76 @@ public object ConvertSourceToIntermediate(IPublishedElement owner, IPublishedPro
 
     return null;
 }
+
+//For simple property editors, like a textbox, that only store a text string,
+//you can just return the raw value since the intermediate and raw values are the same
+public object ConvertSourceToIntermediate(IPublishedElement owner, IPublishedPropertyType propertyType, object source, bool preview)
+{
+    return source as string;
+}
 ```
 
 ### ConvertIntermediateToObject(IPublishedElement owner, IPublishedPropertyType propertyType, PropertyCacheLevel referenceCacheLevel, object inter, bool preview)
 
-This method converts the Intermediate to an Object. The returned value is used by the `GetPropertyValue<T>` method of `IPublishedContent`.
-
-{% include "../../.gitbook/includes/obsolete-warning-ipublishedsnapshotaccessor.md" %}
+This method converts the intermediate value to an object of the type specified in the `GetPropertyValueType()` method of the Property Value Converter. The returned value is used by the `GetPropertyValue<T>` method of `IPublishedContent`.
 
-The below example converts the nodeId (converted to `Int` or `Udi` by _ConvertSourceToIntermediate_) into an 'IPublishedContent' object.
+The example below converts the node ID (converted to `int` or `Udi` by _ConvertSourceToIntermediate_) into an `IPublishedContent` object.
 
 ```csharp
+// In this example _contentCache is an instance of IPublishedContentCache that is injected
```

Comment:
If we use the implementation from the full sample here, replace the code below with this.

```
// Converts the intermediate value (GuidUdi) to the actual object value (IPublishedContent)
public object? ConvertIntermediateToObject(IPublishedElement owner, IPublishedPropertyType propertyType,
	PropertyCacheLevel referenceCacheLevel, object? inter, bool preview)
	=> inter is GuidUdi guidUdi
		? publishedContentCache.GetById(guidUdi.Guid)
		: null;
```